### PR TITLE
Don't Notify Users Of Declined Charges on Frozen Cards

### PR DESCRIPTION
**Pull request by sampoder**
I've probably done this wrong but this would be sick to have.... I get about 100 of these each week:

Screenshot 2023-02-17 at 9 39 31 PM

Once, I got 200 in a day.

### CODE REVIEW

File: app/services/stripe_authorization_service/create_from_webhook.rb [hunk 1]
```diff
@@ -39,7 +39,8 @@ def run
           end
           # rubocop:enable Naming/VariableNumber
         else
-          CanonicalPendingTransactionMailer.with(canonical_pending_transaction_id: cpt.id).notify_declined.deliver_later
+          if cpt.raw_pending_stripe_transaction.stripe_transaction["request_history"][0]["reason"] != "card_inactive":
```
Comment (sampoder) [inline app/services/stripe_authorization_service/create_from_webhook.rb]:
```suggestion
          if cpt.raw_pending_stripe_transaction.stripe_transaction["request_history"][0]["reason"] != "card_inactive"
```

File: app/services/stripe_authorization_service/create_from_webhook.rb [hunk 2]
```diff
@@ -39,7 +39,8 @@ def run
           end
           # rubocop:enable Naming/VariableNumber
         else
-          CanonicalPendingTransactionMailer.with(canonical_pending_transaction_id: cpt.id).notify_declined.deliver_later
+          if cpt.raw_pending_stripe_transaction.stripe_transaction["request_history"][0]["reason"] != "card_inactive"
```
Comment (cjdenio) [inline app/services/stripe_authorization_service/create_from_webhook.rb]: This includes cards that haven't been activated yet, which I think decline emails should still be sent for. You might change this to `unless cpt&.stripe_card&.frozen?` and update [`frozen?`](https://github.com/hackclub/bank/blob/8ec63112665d3ea59b4750b24f4fa4680d505648/app/models/stripe_card.rb#L178-L180) to be
```ruby
def frozen?
  activated? && stripe_status == "inactive"
end
```

Comment (sampoder) [reply app/services/stripe_authorization_service/create_from_webhook.rb]: sure!

Comment (sampoder) [reply app/services/stripe_authorization_service/create_from_webhook.rb]: done :D

File: app/services/stripe_authorization_service/create_from_webhook.rb [hunk 3]
```diff
@@ -39,7 +39,8 @@ def run
           end
           # rubocop:enable Naming/VariableNumber
         else
-          CanonicalPendingTransactionMailer.with(canonical_pending_transaction_id: cpt.id).notify_declined.deliver_later
+          unless cpt&.stripe_card&.frozen?
+            CanonicalPendingTransactionMailer.with(canonical_pending_transaction_id: cpt.id).notify_declined.deliver_later
```
Comment (sampoder) [inline app/services/stripe_authorization_service/create_from_webhook.rb]:
```suggestion
            CanonicalPendingTransactionMailer.with(canonical_pending_transaction_id: cpt.id).notify_declined.deliver_later
          end
```

**Review by cjdenio — APPROVED** [review]
Thanks!

**Pull request merged**